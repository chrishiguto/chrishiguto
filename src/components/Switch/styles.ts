import styled, { css } from 'styled-components'

export const Wrapper = styled.button`
  ${({ theme }) => css`
    cursor: pointer;
    outline: none;
    border: none;

    width: 4.4rem;
    height: 2.4rem;
    background-color: ${theme.colors.gray};
    border-radius: 2.2rem;
  `}
`

type CircleProps = {
  checked: boolean
}

const circleModifiers = {
  checked: () => css`
    transform: translateX(2.2rem);
  `
}

export const Circle = styled.div<CircleProps>`
  ${({ theme, checked }) => css`
    display: flex;
    align-items: center;
    justify-content: center;

    width: 2.4rem;
    height: 2.4rem;
    background-color: ${theme.colors.white};
    border-radius: 100%;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.3);
    transition: all 0.4s;

    ${checked && circleModifiers.checked()}
  `}
`
