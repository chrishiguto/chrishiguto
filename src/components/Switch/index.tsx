import * as S from './styles'

export type SwitchProps = {
  checked: boolean
  handleClick: () => void
  activeIcon?: React.ReactNode
  inactiveIcon?: React.ReactNode
}

const Switch = ({
  checked,
  handleClick,
  activeIcon,
  inactiveIcon
}: SwitchProps) => (
  <S.Wrapper onClick={handleClick}>
    <S.Circle checked={checked}>
      {!!activeIcon && checked && activeIcon}
      {!!inactiveIcon && !checked && inactiveIcon}
    </S.Circle>
  </S.Wrapper>
)

export default Switch
