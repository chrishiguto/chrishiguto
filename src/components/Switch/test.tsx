import { render, screen } from '@testing-library/react'

import Switch from '.'

describe('<Switch />', () => {
  it('should render the heading', () => {
    const { container } = render(<Switch />)

    expect(screen.getByRole('heading', { name: /Switch/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
