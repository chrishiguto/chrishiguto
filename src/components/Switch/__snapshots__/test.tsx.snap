// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Switch
  </h1>
</main>
`;
